### PR TITLE
[DEV-342] Add unit tests

### PR DESCRIPTION
**Pull request by grp06**

### CODE REVIEW

**Comment by grp06** [comment]
I also downgraded to Solidity 0.8.9 because that is the latest version supported by Hardhat: https://hardhat.org/hardhat-runner/docs/reference/solidity-support

**Comment by grp06** [comment]
This PR is not ready to merge. So far I just have the existing tests from ERC4973's repo

To-do
- get `vm.expectEmit` functionality working
-  write unit tests for functions unique to `Badges.sol`
- research `_disableInitializers()` and see if we need it in our contracts. If so, figure out how to make tests/initializers work with it

File: src/SpecDataHolder.t.sol [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { Badges } from "./Badges.sol";
+import { SpecDataHolder } from "./SpecDataHolder.sol";
+import { Raft } from "./Raft.sol";
+
+contract SpecDataHolderTest is Test {
+  Raft raft;
+  Badges badges;
+  SpecDataHolder specDataHolder;
+
+  address passiveAddress = 0x0f6A79A579658E401E0B81c6dde1F2cd51d97176;
```
Comment (rahulrumalla) [inline src/SpecDataHolder.t.sol:18]: Could we move all ofthese into environment variables and read from there?
https://book.getfoundry.sh/cheatcodes/external

File: src/Badges.t.sol [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+    rejecter = new ERC1271Mock(false);
+    aa = new AccountAbstraction(true);
+
+    specUri = "some spec uri";
+
+    vm.label(passiveAddress, "passive");
+  }
+
+  // helper function
+  function createRaftAndRegisterSpec() internal {
```
Comment (rahulrumalla) [inline src/Badges.t.sol:98]: Move internal funcs to after public funcs per styleguide
https://docs.soliditylang.org/en/latest/style-guide.html#order-of-functions

**Review by rahulrumalla — COMMENTED** [review]
Left some comments

**Pull request merged**